Add meta stuff to submit and detail views

Oksamies opened the pull request:
refs TS-1077

anttimaki commented on apps/sertra/components/HeadWrapper.tsx:
```diff
@@ -0,0 +1,18 @@
+
+export const HeadWrapper: React.FC<HeadProps> = ({ title, description }) => (
+  <Head>
+    <title>{title}</title>
```
This could be ```{`${title} | Thunderstore`}``` so we don't need to repeat it everywhere.

Oksamies replied:
True, I'll add it there 👍

anttimaki commented on apps/sertra/pages/create.tsx:
```diff
@@ -6,6 +6,7 @@ import { getPackageList } from "../api/hooks";
 import { ApiURLs, TsApiURLs } from "../api/urls";
 import { ModSelectorModal } from "../components/mod-selector/modal";
 import styles from "../styles/SubmitServer.module.css";
+import { HeadWrapper } from "../components/HeadWrapper";
```
Import order plz.

anttimaki commented on apps/sertra/pages/server/[id].tsx:
```diff
@@ -13,6 +13,7 @@ import { FetchListingData } from "../../api/PackageDataFetcher";
 import { ServerInfo } from "../../components/ServerInfo";
 import { ServerInstructions } from "../../components/ServerInstructions";
 import styles from "../../styles/ServerDetail.module.css";
+import { HeadWrapper } from "../../components/HeadWrapper";
```
Import order plz.

anttimaki commented on apps/sertra/pages/server/[id].tsx:
```diff
@@ -51,6 +52,10 @@ const ServerDetail: React.FC<{
   return (
     <div className={styles.container}>
+      <HeadWrapper
+        title={`${listing_data.name} - V-Rising Servers | Thunderstore`}
```
Could we read the game name from `listing_data`?

Oksamies replied:
Yes but it's lower case, I'll add it from there tho. Better to have dynamically.

anttimaki commented on apps/sertra/pages/server/[id].tsx:
```diff
@@ -51,6 +52,10 @@ const ServerDetail: React.FC<{
     <div className={styles.container}>
+      <HeadWrapper
+        title={`${listing_data.name} - V-Rising Servers | Thunderstore`}
+        description={listing_data.description}
```
Just a thought, but are we certain we want to use user inputs (description here, name above) as the meta content? E.g. racist content is obviously bad when shown on our website, but can be extra bad when shown on those embeds on other sites?

Oksamies replied:
This was a quick suggest by @MythicManiac , I don't have a good understanding/knowledge what should/could be used here. What do you think @MythicManiac ?

MythicManiac replied:
That's the purpose of the meta fields, we'll just need to deal with moderating them. The alternative is not having any content-bound info available in any embeds, which is also bad.

anttimaki commented:
Also I think the commit message is bad. What is "meta stuff"? Why isn't the header wrapper mentioned? Now that I think of it, should it be documented in README oslt so other devs know to use it?

Oksamies commented:
> Also I think the commit message is bad. What is "meta stuff"? Why isn't the header wrapper mentioned? Now that I think of it, should it be documented in README oslt so other devs know to use it?

Fixed comments and added a bit of information to the readme.

I'll be waiting for a answer from @MythicManiac to the matter of using user inputs.
